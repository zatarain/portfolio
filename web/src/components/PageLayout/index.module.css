.container {
	max-width: 1440px;
	margin: 0 auto;
	text-align: left;
	background: var(--bg-page);
	overflow: hidden;
}

.hero {
	margin: 2rem auto;
	text-align: center;
	position: relative;
	overflow: hidden;
	top: -2rem;
	max-width: 1440px;
	max-height: 1080px;
}

.hero::before {
	content: '';
	position: absolute;
	z-index: 2;
	height: 100%;
	bottom: 0;
	left: 0;
	right: 0;
	box-shadow: 0 -7rem 7rem 7rem var(--bg-primary) inset;
	clip-path: ellipse(100% 50% at center 42.6%);
}

.hero img {
	display: inline-block;
	position: relative;
	clip-path: ellipse(100% 50% at center 42.5%);
}

.carousel .control-dots {
	bottom: 4rem;
	z-index: 3;
}
