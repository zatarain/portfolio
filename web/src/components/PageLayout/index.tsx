import Head from 'next/head';
import { ReactNode } from 'react';
import { Inter } from 'next/font/google'
import Image from 'next/image';
import Link from 'next/link'
import HeroSlider from '#components/HeroSlider'

interface Properties {
	children?: ReactNode
	data: any
	title?: string
}

const images = {
	Liverpool: '002',
	Germany: '003',
	Netherlands: '004',
	Norway: '005',
	Scotland: '006',
}

const inter = Inter({ subsets: ['latin'] })

export default function PageLayout({ children, title, data }: Properties) {
	const pageTitle = [data?.name, title].filter(part => !!part).join(' - ')
	return (
		<div className={`${inter.className} container`}>
			<Head>
				<title>{pageTitle}</title>
				<link rel="icon" href="/favicon.ico" />
			</Head>
			<header>
				<nav>
					<div>
						<div className="pages">
							<Link href="/" className="logo">
								<Image alt="logo" src="/logo.svg" width="48" height="48" />
								{data?.name}
							</Link>
							<div className="sections">
								<Link href="/">Home</Link>
								<Link href="#">Experience</Link>
								<Link href="#">Education</Link>
								<Link href="#">Skills</Link>
								<Link href="#">Projects</Link>
								<Link href="#">Hobbies</Link>
								<Link href="#">Contact</Link>
							</div>
						</div>
						<div className="actions">
							<Link href="#">E-mail</Link>
							<Link href="#" className="call-to-action">Download</Link>
						</div>
					</div>
				</nav>
				<HeroSlider images={images} />
			</header>
			<main>
				{children}
			</main>
			<footer>
				<ul>
					<li>
						<h4>Tech Stuff</h4>
						<ul>
							<li><a href="#">C++</a></li>
							<li><a href="#">TypeScript</a></li>
							<li><a href="#">Ruby</a></li>
							<li><a href="#">Terraform</a></li>
							<li><a href="#">Golang</a></li>
							<li><a href="#">Python</a></li>
							<li><a href="#">Cloud - AWS</a></li>
							<li><a href="#">Git</a></li>
						</ul>
					</li>
					<li>
						<h4>Skills</h4>
						<ul>
							<li><a href="#">Algorithms and Data Structures</a></li>
							<li><a href="#">Mathematics</a></li>
							<li><a href="#">Software Engineering</a></li>
							<li><a href="#">Web Development</a></li>
							<li><a href="#">Continuous Integration</a></li>
							<li><a href="#">Tech Agnosticity</a></li>
							<li><a href="#">Proactivity</a></li>
							<li><a href="#">Continuous Learning</a></li>
						</ul>
					</li>
					<li>
						<h4>Personal Projects</h4>
						<ul>
							<li><a href="#">Algorists Group</a></li>
							<li><a href="#">Unit Test Zeal</a></li>
							<li><a href="#">Personal Website</a></li>
							<li><a href="#">Jarnik Build System</a></li>
							<li><a href="#">Spectral Clustering</a></li>
						</ul>
					</li>
					<li>
						<h4>Volunteering</h4>
						<ul>
							<li><a href="#">LEGO Robotics and Tech for Kids</a></li>
							<li><a href="#">Plant Trees</a></li>
							<li><a href="#">OmegaUp Latinoamerica</a></li>
							<li><a href="#">Rider Photoshoot</a></li>
							<li><a href="#">Charity Events</a></li>
						</ul>
					</li>
					<li>
						<h4>Hobbies and fun</h4>
						<ul>
							<li><a href="#">Cooking</a></li>
							<li><a href="#">Liverpool FC</a></li>
							<li><a href="#">Anime</a></li>
							<li><a href="#">Videogames</a></li>
							<li><a href="#">Boardgames</a></li>
							<li><a href="#">Watching sports</a></li>
							<li><a href="#">Travel and Hiking</a></li>
						</ul>
					</li>
					<li>
						<h4>Hates and boring</h4>
						<ul>
							<li><a href="#">Apple stuff</a></li>
							<li><a href="#">Real Madrid FC</a></li>
							<li><a href="#">Java stuff</a></li>
						</ul>
					</li>
				</ul>
			</footer>
		</div>
	)
}
