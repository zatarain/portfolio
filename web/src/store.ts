import { configureStore, ThunkAction, Action, createSlice } from '@reduxjs/toolkit'

export function makeStore() {
  const responsiveFlipper = createSlice({
    name: 'flipper',
    initialState: {
      responsive: false,
    },
    reducers: {
      flipResponsive: (state) => {
        state.responsive = !state.responsive
      }
    }
  })
  return configureStore({
    reducer: { flipper: responsiveFlipper.reducer },
  })
}

const store = makeStore()

export type AppState = ReturnType<typeof store.getState>

export type AppDispatch = typeof store.dispatch

export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  AppState,
  unknown,
  Action<string>
>

export default store
