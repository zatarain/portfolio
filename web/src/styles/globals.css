@import url('https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css');
@import url('https://fonts.cdnfonts.com/css/cascadia-code');
@import url('react-responsive-carousel/lib/styles/carousel.min.css');
@import url('leaflet/dist/leaflet.css');


:root {
  --bg-page: #fafbfc;
  --bg-primary: #28b873;
  --bg-secondary: #151b28;
  --bg-action: #f40;
  --bg-action-hover: #f60;
  --link-secondary: #a5acba;
  --highlight-secondary: #5cdeac;
  --fg-secondary: #fff;
  --fg-menuitem-hover: #7fffd4;
  --ldn-dark: #29bc9b;
  --ldn-lighter: #aaffec;
  --ldn-light: #79ffe1;
  --ldn: #50e3c2;
  --heading-2: var(--ldn-dark);
  --heading-3: var(--ldn-dark);
  --headings-font: Cambria, Cochin, Georgia, Times, 'Times New Roman', serif;
}

@media screen {
  html, body {
    padding: 0;
    margin: 0;
    text-align: center;
    font-family: Inter, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen, Ubuntu,
      Cantarell, Fira Sans, Droid Sans, Helvetica Neue, sans-serif;
    font-size: 14px;
    scroll-behavior: smooth;
  }

  main {
    padding: 0 3rem;
    margin: 3rem 0;
    font-size: 1.2rem;
    line-height: 1.75rem;
  }
}

* {
  box-sizing: border-box;
}

a:link, a:visited {
  color: var(--ldn-dark);
  text-decoration: none;
}

a:hover, a:hover:visited {
  color: var(--bg-action);
}

.carousel ul.control-dots {
	bottom: 4rem;
	z-index: 3;
}

.carousel ul.control-dots li {
  width: 0.5rem;
  height: 0.5rem;
  padding: 0.5rem;
}

.carousel figcaption {
  margin: 0;
  position: relative;
  bottom: 3rem;
  font-size: 1rem;
}

.leaflet-bottom,
.leaflet-control,
.leaflet-pane,
.leaflet-top {
  z-index: 0 !important;
}

@media screen and (max-width: 850px) {
  .carousel figure img {
    height: 50vh;
  }

  .carousel figcaption {
    bottom: 1rem;
  }

  .carousel ul.control-dots {
    bottom: 3rem;
  }
}

@media screen and (max-width: 539px) {
  .carousel figure img {
    height: 30vh;
  }

  .carousel figcaption {
    bottom: 3rem;
    display: none;
  }

  .carousel ul.control-dots {
    bottom: -1rem;
  }
}
