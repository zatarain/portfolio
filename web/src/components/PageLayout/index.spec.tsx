import { jest } from '@jest/globals';
import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import store from '#store'
import PageLayout from '.'
import Head from 'next/head'

jest.mock('next/font/google', () => {
	return {
		__esModule: true,
		Inter: jest.fn(() => {
			return {
				className: 'my-font',
			}
		}),
	}
})

jest.mock('next/head');

describe('<PageLayout ...>...</PageLayout>', () => {
	it('renders the component correctly with passed name', async () => {
		const head = jest.mocked(Head)

		const { asFragment, getByText } = render(
			<Provider store={store}>
				<PageLayout title="Test title" data={{ name: 'My Name' }}>
					<p>Hello World</p>
				</PageLayout>
			</Provider>
		)

		expect(head).toBeCalled()
		head.mock.lastCall?.map((tag) => tag.children && render(tag.children, { container: document.head }))

		expect(document.title).toBe('My Name - Test title')
		expect(getByText('My Name')).toBeInTheDocument()
		expect(getByText('Hello World')).toBeInTheDocument()
		expect(asFragment()).toMatchSnapshot()
	})
})
