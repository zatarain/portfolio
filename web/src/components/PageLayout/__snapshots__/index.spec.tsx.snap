// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageLayout ...>...</PageLayout> renders the component correctly with passed name 1`] = `
<DocumentFragment>
  <div
    class="my-font container"
  >
    <header
      class="header"
    >
      <nav
        class="navigation-bar "
        id="navigation-bar"
        role="menubar"
      >
        <a
          class="logo"
          href="/"
        >
          <img
            alt="logo"
            data-nimg="1"
            decoding="async"
            height="48"
            loading="lazy"
            src="/logo.svg"
            style="color: transparent;"
            width="48"
          />
          My Name
        </a>
        <ul
          class="sections"
        >
          <li>
            <a
              href="#work-experience"
            >
              Experience
            </a>
          </li>
          <li>
            <a
              href="#education"
            >
              Education
            </a>
          </li>
          <li>
            <a
              href="#"
            >
              Skills
            </a>
          </li>
          <li>
            <a
              href="#academic-projects"
            >
              Projects
            </a>
          </li>
          <li>
            <a
              href="#volunteering"
            >
              Hobbies
            </a>
          </li>
          <li
            class="action"
          >
            <a
              href="#"
            >
              E-mail
            </a>
          </li>
          <li
            class="action"
          >
            <a
              class="call-to-action"
              href="#"
            >
              Download
            </a>
          </li>
          <li
            class="dropdown"
          >
            <a
              href="#"
              role="menuitemcheckbox"
            >
              <i
                class="fa fa-bars"
              />
            </a>
          </li>
        </ul>
      </nav>
    </header>
    <main>
      <p>
        Hello World
      </p>
    </main>
    <footer
      class="footer"
    >
      <ul>
        <li>
          <h4>
            Tech Stuff
          </h4>
          <ul>
            <li>
              <a
                href="#"
              >
                C++
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                TypeScript
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Ruby
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Terraform
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Golang
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Python
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Cloud - AWS
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Git
              </a>
            </li>
          </ul>
        </li>
        <li>
          <h4>
            Skills
          </h4>
          <ul>
            <li>
              <a
                href="#"
              >
                Algorithms and Data Structures
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Mathematics
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Software Engineering
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Web Development
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Continuous Integration
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Tech Agnostic
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Proactivity
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Continuous Learning
              </a>
            </li>
          </ul>
        </li>
        <li>
          <h4>
            Personal Projects
          </h4>
          <ul>
            <li>
              <a
                href="#"
              >
                Algorists Group
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Unit Test Zeal
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Personal Website
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Jarnik Build System
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Spectral Clustering
              </a>
            </li>
          </ul>
        </li>
        <li>
          <h4>
            Volunteering
          </h4>
          <ul>
            <li>
              <a
                href="#"
              >
                LEGO Robotics and Tech for Kids
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Plant Trees
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                OmegaUp Latinoamerica
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Rider Photoshoot
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Charity Events
              </a>
            </li>
          </ul>
        </li>
        <li>
          <h4>
            Hobbies and fun
          </h4>
          <ul>
            <li>
              <a
                href="#"
              >
                Cooking
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Liverpool FC
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Anime
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Videogames
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Boardgames
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Watching sports
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Travel and Hiking
              </a>
            </li>
          </ul>
        </li>
        <li>
          <h4>
            Hates and boring
          </h4>
          <ul>
            <li>
              <a
                href="#"
              >
                Apple stuff
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Real Madrid FC
              </a>
            </li>
            <li>
              <a
                href="#"
              >
                Java stuff
              </a>
            </li>
          </ul>
        </li>
      </ul>
    </footer>
  </div>
</DocumentFragment>
`;
